1.2/percentile_test.go

```diff
@@ -44,10 +44,8 @@ func TestGeneratePercentileSecond(t *testing.T) {
 	data := []uint{16, 6, 7, 8, 8, 10, 13, 15, 3, 20}
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("PASS"), func(t *testing.T) {
-			if output, err := generatePercentile(test.input); err != nil {
-				t.Errorf("expected no error, got: %v", err)
-			} else if output(data) != test.p {
-				t.Errorf("Fail expected: %v, got: %v", test.p, output)
+			if output := Must(generatePercentile, test.input); output(data) != test.p {
+				t.Errorf("Fail expected: %v, got: %v", test.p, output(data))
```

Comment:
nit: You call output twice here... not the worst thing, but it is an O(n lg n) operation.

Reply:
Changed to `if output := mustGeneratePercentile(test.input)(data)` and now output is the result of the inner percentile function